lib/src/importer-registry.ts

```diff
@@ -43,10 +50,26 @@ export class ImporterRegistry<sync extends 'sync' | 'async'> {
 
   /** Converts an importer to a proto without adding it to `this.importers`. */
   register(
-    importer: Importer<sync> | FileImporter<sync>
+    importer: Importer<sync> | FileImporter<sync> | NodePackageImporter
   ): proto.InboundMessage_CompileRequest_Importer {
     const message = new proto.InboundMessage_CompileRequest_Importer();
-    if ('canonicalize' in importer) {
+    if (importer instanceof NodePackageImporter) {
+      const importerMessage = new proto.NodePackageImporter();
+      const entryPointPath = importer.entryPointPath
+        ? p.resolve(process.cwd(), importer.entryPointPath)
+        : require.main?.filename;
+
+      if (!entryPointPath) {
+        throw new Error(
+          'Node Package Importer requires access to a filesystem.'
```

Comment:
Don't title-case "Node package importer", and provide more information about _why_ this is failing. Explain that `require.main.filename` is unset.

Reply:
Fixed in https://github.com/oddbird/embedded-host-node/commit/b8e96d64c1343d73d3ab053c00f5c8b3f94b77bf